src/utils/skip_list.hpp

```diff
@@ -591,13 +591,20 @@ class SkipList final : detail::SkipListNode_base {
     Iterator(TNode *node) : node_(node) {}
 
    public:
-    TObj &operator*() const { return node_->obj; }
+    using value_type = TObj;
+    using difference_type = std::ptrdiff_t;
 
-    TObj *operator->() const { return &node_->obj; }
+    Iterator() = default;
```

Comment:
Shouldn't this be deleted or private?
We don't handle `node_ == nullptr` so lets make `Iterator(TNode *node)` the only constructor.

Reply:
Perhaps you actually wanted to do a destructor, since we are missing it.